paddleseg/models/pfpnnet.py

```diff
@@ -0,0 +1,213 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+
+import paddle
+import paddle.nn as nn
+import paddle.nn.functional as F
+
+from paddle.nn import layer
+from paddleseg.models import layers
+from paddleseg.cvlibs import manager
+from paddleseg.utils import utils
+from paddle.fluid.layers.nn import scale
+
+
+@manager.MODELS.add_component
+class PFPNNet(nn.Layer):
+    """
+    The Panoptic Feature Pyramid Networks implementation based on PaddlePaddle.
+
+    The original article refers to
+    Alexander Kirillov, Ross Girshick, Kaiming He, Piotr Dollár, et al. "Panoptic Feature Pyramid Networks"
+    (https://arxiv.org/abs/1901.02446)
+
+    Args:
+        num_classes (int): The unique number of target classes.
+        backbone (Paddle.nn.Layer): Backbone network, currently support Resnet50/101.
+        backbone_indices (tuple): Four values in the tuple indicate the indices of output of backbone.
+        enable_auxiliary_loss (bool, optional): A bool value indicates whether adding auxiliary loss. Default: False.
+        align_corners (bool, optional): An argument of F.interpolate. It should be set to False when the feature size is even,
+            e.g. 1024x512, otherwise it is True, e.g. 769x769. Default: False.
+        pretrained (str, optional): The path or url of pretrained model. Default: None.
+    """
+    def __init__(self,
+                 num_classes,
+                 backbone,
+                 backbone_indices,
+                 channels,
+                 enable_auxiliary_loss=False,
+                 align_corners=False,
+                 dropout_ratio=0.1,
+                 fpn_inplanes=[256, 512, 1024, 2048],
+                 pretrained=None):
+        super(PFPNNet, self).__init__()
+        self.backbone = backbone
+        self.backbone_indices = backbone_indices
+        self.in_channels = [
+            self.backbone.feat_channels[i] for i in backbone_indices
+        ]
+        self.align_corners = align_corners
+        self.pretrained = pretrained
+        self.enable_auxiliary_loss = enable_auxiliary_loss
+
+        self.head = PFPNHead(num_class=num_classes,
+                             fpn_inplanes=fpn_inplanes,
+                             dropout_ratio=dropout_ratio,
+                             channels=channels,
+                             fpn_dim=channels,
+                             enable_auxiliary_loss=self.enable_auxiliary_loss)
+        self.init_weight()
+
+    def forward(self, x):
+        feats = self.backbone(x)
+        feats = [feats[i] for i in self.backbone_indices]
+        logit_list = self.head(feats)
+        logit_list = [
+            F.interpolate(logit,
+                          paddle.shape(x)[2:],
+                          mode='bilinear',
+                          align_corners=self.align_corners)
+            for logit in logit_list
+        ]
+        return logit_list
```

Comment:
You can 

return [F.interpolate ... ] 

instead.

Reply:
fix